### PR TITLE
Add real `on_failure` page

### PR DESCRIPTION
**Pull request by kiview**
When Testcontainers fails to automatically discover a supported container runtime environments, it will log a human readable error message with a link, pointing to https://www.testcontainers.org/supported_docker_environment/. In order to give new users with less Docker experience more suitable hints on how to fix their issue, we intend to explore ways to make the documentation more streamlined.

### CODE REVIEW

File: docs/_redirects [hunk 1]
```diff
@@ -12,7 +12,7 @@
 /usage/database_containers.html         /modules/databases/
 /usage/neo4j_container.html             /modules/databases/neo4j/
 /compatibility.html                     /supported_docker_environment/
-/on_failure.html                        /supported_docker_environment/
+/on_failure.html                        /on_failure
```
Comment (mtremsal) [inline docs/_redirects:15]: Let's have a more descriptive url. e.g. `error_missing_container_runtime_environment.html`

File: docs/on_failure.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Fixing Issues with Discovering A Supported Container Runtime Environment
+
+If you ended up on this page, 
+it seems that either Testcontainers was not able to find a supported container runtime in your environment,
+or you found page while browsing the documentation in search of information on dealing with errors regarding Testcontainers' environment discovery mechanism.
```
Comment (mtremsal) [inline docs/on_failure.md:5]: or you found _this_ page while searching for information to deal with errors regarding the environment discovery mechanism of Testcontainers.

File: docs/on_failure.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+Testcontainers requires a supported container runtime environment to be present in order to function.
```
Comment (mtremsal) [inline docs/on_failure.md:7]: Testcontainers requires a supported container runtime environment to be present in order to execute containers.

File: docs/on_failure.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+Here is a list of supported container runtime environments, that can get you up to speed quickly:
```
Comment (mtremsal) [inline docs/on_failure.md:8]: Here is a list of supported container runtime environments:

File: docs/on_failure.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+* [Docker Desktop](https://www.docker.com/products/docker-desktop/)
+* [Docker Engine on Linux](https://docs.docker.com/engine/install/)
+* [Testcontainers Cloud](https://www.testcontainers.cloud)
+
+For more extensive information on supported container runtime environments, and how to configure container runtime environments that are not supported by default,
```
Comment (mtremsal) [inline docs/on_failure.md:14]: For more extensive information on supported container runtime environments, as well as known limitations of alternative container runtime environments, please refer to...

**Review by mtremsal — CHANGES_REQUESTED** [review]
Some suggestions to clean-up the language a bit.

**Review by mtremsal — APPROVED** [review]
LGTM -- thanks for the changes 👍

**Comment by kiview** [comment]
Thanks for the review @mtremsal and thanks for applying the changes @shelajev 👍

**Pull request merged**